contracts/modules/metadata/CoreMetadataViewModule.sol

```diff
@@ -0,0 +1,143 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity ^0.8.23;
+
+import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import { Strings } from "@openzeppelin/contracts/utils/Strings.sol";
+import { Base64 } from "@openzeppelin/contracts/utils/Base64.sol";
+import { BaseModule } from "../BaseModule.sol";
+import { IIPAccount } from "../../interfaces/IIPAccount.sol";
+import { IPAccountStorageOps } from "../../lib/IPAccountStorageOps.sol";
+import { CORE_METADATA_VIEW_MODULE_KEY, CORE_METADATA_MODULE_KEY } from "../../lib/modules/Module.sol";
+import { ICoreMetadataViewModule, IViewModule } from "../../interfaces/modules/metadata/ICoreMetadataViewModule.sol";
+import { IModuleRegistry } from "contracts/interfaces/registries/IModuleRegistry.sol";
+
+/// @title Implementation of the ICoreMetadataViewModule interface
+/// @dev Provides functionalities to retrieve core metadata of IP assets, including name, description, and more.
+contract CoreMetadataViewModule is BaseModule, ICoreMetadataViewModule {
+    using IPAccountStorageOps for IIPAccount;
+
+    string public constant override name = CORE_METADATA_VIEW_MODULE_KEY;
+
+    address public immutable IP_ASSET_REGISTRY;
+    address public immutable MODULE_REGISTRY;
+
+    address public coreMetadataModule;
+
+    constructor(address ipAssetRegistry, address moduleRegistry) {
+        IP_ASSET_REGISTRY = ipAssetRegistry;
+        MODULE_REGISTRY = moduleRegistry;
+    }
+
+    /// @notice Updates the address of the CoreMetadataModule used by this view module.
+    /// @dev Retrieve the address of the CoreMetadataModule from the ModuleRegistry.
+    function updateCoreMetadataModule() external {
+        coreMetadataModule = IModuleRegistry(MODULE_REGISTRY).getModule(CORE_METADATA_MODULE_KEY);
+    }
+
+    /// @inheritdoc ICoreMetadataViewModule
```

Comment:
Can you use full natspec instead of inheritdoc, please? Keeping the natspec usage consistent throughout our codebase.

Reply:
Changed to full NatSpec.